src/main/java/org/spongepowered/api/data/type/RailDirection.java

```diff
@@ -26,9 +26,34 @@
 
 import org.spongepowered.api.CatalogType;
 import org.spongepowered.api.util.Cycleable;
+import org.spongepowered.api.util.Direction;
 import org.spongepowered.api.util.annotation.CatalogedBy;
 
+import java.util.Optional;
+
 @CatalogedBy(RailDirections.class)
 public interface RailDirection extends CatalogType, Cycleable<RailDirection> {
 
+    /**
+     * Returns the ascending direction, if this rail direction is ascending.
+     *
+     * @return The ascending direction, if possible
+     */
+    Optional<Direction> getAscendingDirection();
```

Comment:
Maybe instead of using an `Optional`, we should use a `boolean` for `isAscending`/`isSlope`, because there is no `DOWN` here.

Reply:
Never mind, I misunderstood the method. Maybe clarify it a bit in the javadocs, that this is the direction it's ascending to.